src/addons/model.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @fileoverview A general modelling class used by the Presenter's
+ * getModel function.
+ */
+
+import { set, Emitter } from '../microcosm'
+
+function isObservable(binding) {
+  return binding && typeof binding.subscribe === 'function'
+}
+
+function isCallable(binding) {
+  return binding && typeof binding.call === 'function'
+}
+
+function invoke(binding, repo, scope) {
+  if (isCallable(binding)) {
+    return binding.call(scope, repo.state, repo)
+  }
+
+  return binding
+}
+
+export default class Model extends Emitter {
+  /**
+   * @param {Microcosm} repo Track this Microcosm instance for updates
+   * @param {scope} scope Scope to invoke functional bindings
+   */
+  constructor(repo, scope) {
+    super()
+
+    this.repo = repo
+    this.scope = scope
+    this.bindings = {}
+    this.subscriptions = {}
+    this.value = {}
+
+    this.repo.on('change', this.compute, this)
+  }
+
+  /**
+   * Track an observable. Sending updates to a given key.
+   * @param {string} key
+   * @param {Observable} observable
+   */
+  track(key, observable) {
+    let last = this.subscriptions[key]
+    let next = observable.subscribe(value => this.set(key, value))
+
+    this.subscriptions[key] = next
+
+    if (last) {
+      last.unsubscribe()
+    }
+  }
+
+  /**
+   * @param {Object} bindings An set of key/value pairs for building a model
+   */
+  bind(bindings) {
+    this.bindings = {}
+
+    for (var key in bindings) {
+      let binding = bindings[key]
+
+      if (isObservable(binding)) {
+        this.track(key, binding)
+      } else {
+        this.bindings[key] = binding
+      }
+    }
+
+    this.compute()
+  }
```

Comment:
1. When a presenter gets new props or state, we invoke `getModel`. This is so that you can create bindings specific to a particular prop. This happens quite a bit for URL parameters:

```javascript
getModel (props) {
  const { params } = props

  return {
    user: state => state.users.find(user => user.id === params.id
  }
}
```

getModel is not recalculated if repo state changes. Only presenter props/state.

2. I don't like the name `bind` either. This is essentially updating the model with a fresh set of data subscriptions. Any suggestions?

---

Beyond that, this method of building a view model comes from Redux's `Connect/Provide` pattern. For a while, I've felt like it's not very efficient. You execute the model callbacks every time, relying on immutable data to make it fast. This isn't good enough, a lot of the times we run `array.filter` or something, which always returns a new array, causing an unnecessary render. You can avoid this for a primary data calculation, but you almost always get into trouble if you have some sort of secondary calculation. 

So something to chew on. This is part of the reason I want to move towards observables.